app/setup_handlers.go

```diff
@@ -22,6 +24,37 @@ func setUpgradeHandler(app *ElysApp) {
 		func(ctx sdk.Context, plan upgradetypes.Plan, vm m.VersionMap) (m.VersionMap, error) {
 			app.Logger().Info("Running upgrade handler for " + version.Version)
 
+			if version.Version == "v0.38.0" || version.Version == "v999.999.999" {
+				// Retrieve the wasm module store key
+				storeKey := app.keys[wasmmodule.StoreKey]
+
+				// Retrieve the wasm module store
+				store := ctx.KVStore(storeKey)
+
+				// List of prefixes to clear
+				prefixes := [][]byte{
+					// Account History contract
+					wasmtypes.GetContractStorePrefix(sdk.MustAccAddressFromBech32("elys17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs98tvuy")),
+				}
+
+				// Add old code keys to the list of prefixes to clear
+				for i := uint64(1); i < 671; i++ {
```

Comment:
are we sure this won't affect the TS functionality ? just confirming

Reply:
what if old contracts make a call to these IDs ? will that halt the chain with invalid mem error ?

Reply:
@amityadav0 a cosmwasm code is the binary of the contract stored on the chain, it is not the actual instance/state of the contract, when we deploy a new version of the contract, we first store the new code of the contract binary (that creates a new code id) and then we migrate the contract from the previous code id to the new one, so any past code ids are unused. AH, TS and FS are using the code IDs 671, 672 and 673 so we are safe.

Reply:
got it, makes sense